routes/stats.go

```diff
@@ -41,20 +41,39 @@ func Stats(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err := db.QueryRow(
-		`SELECT COALESCE(SUM(bytes), 0), COUNT(*)
-		   FROM solutions
-		   JOIN code ON code_id = id
-		  WHERE NOT failing`,
-	).Scan(&data.Bytes, &data.Solutions); err != nil {
+		`WITH distinct_solutions AS (
+		  SELECT DISTINCT ON (hole, lang, user_id, code_id)
+		         hole,
+		         lang,
+		         user_id,
+		         code_id,
+		         bytes
+		    FROM solutions
+		    JOIN code ON code_id = code.id
+		    WHERE NOT failing
+		) SELECT COALESCE(SUM(bytes), 0),
+		         COUNT(*),
+		         COUNT(*) - COUNT(DISTINCT(hole, lang, user_id))
```

Comment:
How about we lose this line, so just have the count of unique solutions (regardless of their "slot") and the sum of their size. Don't worry too much about query performance, I'm thinking the stats page would be a perfect use case for a materialised view in the near future. While this solution gives less information that your current PR, I think it's good enough for an overview page like stats, especially since once we copy the chars solution into the bytes column the number will shoot up.

So
```sql
SELECT COALESCE(SUM(bytes), 0), COUNT(*)
```

Reply:
That sounds fine to me. I just thought it would be interesting to see how many second solutions there are for the same user/hole/lang. I'm not sure what you mean about the number changing when we copy the chars solution though. I would expect the numbers to stay the same.

I think the wording "in second slot" is a bit awkward. Maybe "second solutions" is better? I don't mind dropping it either.

Reply:
Nah you're right, the number would stay the same. I'd rather just show the single count for now, we can revisit adding more information to the stats route after this whole feature ships, thanks.